pages/price-feeds/use-real-time-data/starknet.mdx

```diff
@@ -0,0 +1,210 @@
+
+```toml copy
+[dependencies]
+pyth = { git = "https://github.com/pyth-network/pyth-crosschain.git", tag = "pyth-starknet-contract-v0.1.0"}
+```
+
+Pyth also provides a javascript SDK to interact with the Pyth contract on Starknet. You can install it using the following command:
+
+<Tabs items={["npm", "yarn"]}>
+  <Tabs.Tab>
+    ```sh copy $ npm install --save @pythnetwork/pyth-starknet-js ```
+  </Tabs.Tab>
+  <Tabs.Tab>```sh copy $ yarn add @pythnetwork/pyth-starknet-js ```</Tabs.Tab>
+</Tabs>
+
+## Write Contract Code
+
+The code snippet below provides an example module fetching the ETH/USD price from Pyth price feeds:
+
+```rust {12,17,49,57,62-64} copy
+#[starknet::interface]
+pub trait IFetchPrice<T> {
+    fn example_method(
+        ref self: T, destination: ContractAddress, amount_in_usd: u256, price_update: ByteBuffer
+    );
+}
+
+#[starknet::contract]
+mod example_method {
+    use core::panic_with_felt252;
+    use starknet::{ContractAddress, get_caller_address, get_contract_address};
+    use pyth::{ByteBuffer, IPythDispatcher, IPythDispatcherTrait, UnwrapWithFelt252};
+    use openzeppelin::token::erc20::interface::{IERC20CamelDispatcherTrait, IERC20CamelDispatcher};
+
+    #[storage]
+    struct Storage {
+        pyth_address: ContractAddress,
+        eth_erc20_address: ContractAddress,
+        eth_usd_price_feed_id: u256,
+    }
+
+    #[constructor]
+    fn constructor(
+        ref self: ContractState,
+        pyth_address: ContractAddress,
+        eth_erc20_address: ContractAddress,
+        eth_usd_price_feed_id: u256,
+    ) {
+        self.pyth_address.write(pyth_address);
+        self.eth_erc20_address.write(eth_erc20_address);
+        self.eth_usd_price_feed_id.write(eth_usd_price_feed_id);
+    }
+
+    #[abi(embed_v0)]
+    impl FetchPrice of super::IFetchPrice<ContractState> {
+        fn example_method(
+            ref self: ContractState,
+            destination: ContractAddress,
+            amount_in_usd: u256,
+            pyth_price_update: ByteBuffer
+        ) {
+            let pyth = IPythDispatcher { contract_address: self.pyth_address.read() };
+            let eth_erc20 = IERC20CamelDispatcher {
+                contract_address: self.eth_erc20_address.read()
+            };
+            let caller = get_caller_address();
+            let contract = get_contract_address();
+
+            let pyth_fee = pyth.get_update_fee(pyth_price_update.clone(), eth_erc20.contract_address);
+            if !eth_erc20.transferFrom(caller, contract, pyth_fee) {
+                panic_with_felt252("insufficient allowance for fee");
+            }
+            if !eth_erc20.approve(pyth.contract_address, pyth_fee) {
+                panic_with_felt252("approve failed");
+            }
+
+            pyth.update_price_feeds(pyth_price_update);
+
+            // ETH/USD price feed ID
+            // The complete list of feed IDs is available at https://pyth.network/developers/price-feed-ids
```

Comment:
this comment doesn't seem to apply to the code below

IMO, it may actually be clearer to hardcode the contract addresses & feed ids into this code snippet, because (a) it would clarify to the reader what the values look like, and (b) it would shorten the code snippet.

Reply:
I can hardcode the feed id, but I would prefer not to hardcode the contract address because right not we cannot have the same contract address on testnet and on a local katana node, so it would make local runs and CI more cumbersome.

Reply:
We can hardcode the feed IDs for now.